blueprint/core/blueprint_ViewManager.cpp

```diff
@@ -0,0 +1,239 @@
+/*
+  ==============================================================================
+
+    blueprint_ViewManager.cpp
+    Created: 21 May 2020 15:32:00am
+
+  ==============================================================================
+*/
+
+#include "blueprint_ViewManager.h"
+
+namespace blueprint
+{
+    //TODO: Have the root ShadowView owned by ViewManager or have ReactApplicationRoot own this
+    //      and pass a ViewPair like struct to ViewManager constructor?
+    //      Plausible passing a ViewPair fits better with the existing registerViewType
+    //      pattern?
+    ViewManager::ViewManager(View* rootView)
+        : rootShadowView(std::make_unique<ShadowView>(rootView))
+    {
+        jassert(rootView);
+        rootView->addComponentListener(this);
```

Comment:
I think I'd prefer to leave out the ComponentListener here; I think `ReactApplicationRoot` can maintain this line:

```
void resized() override {
  _viewManager->performShadowTreeLayout();
}
```

and then we don't have the need for the listener here. I lean that way because the component listener creates a bit of a circular dependency: the appRoot depends on the view manager, and the view manager depends backwards on the appRoot. (Of course in some sense that dep is always going to be there because the view manager needs to lay out the appRoot and everything else, but this feels like a way to minimize that circular dependency)

Reply:
Yeah cool. I wasn't too sure if only the ViewManager should have the responsibility of triggering performShadowTreeLayout. Will get this fixed up in next set of changes.

Reply:
@nick-thompson resolved in https://github.com/nick-thompson/blueprint/pull/46/commits/3500dd553b88731732f2d6d08f9563c38305aa5b